arviz/stats/diagnostics.py

```diff
@@ -83,7 +215,6 @@ def _get_ess(sample_array):
         raise TypeError("Effective sample size calculation requires multiple chains.")
 
     acov = np.asarray([_autocov(sample_array[chain]) for chain in range(n_chain)])
-
     chain_mean = sample_array.mean(axis=1)
     chain_var = acov[:, 0] * n_draws / (n_draws - 1.0)
     acov_t = acov[:, 1] * n_draws / (n_draws - 1.0)
```

Comment:
Is this correct?

Reply:
It is used in here

rho_hat_odd = 1.0 - (mean_var - np.mean(acov_t)) / var_plus

Reply:
It is not scaled here

https://github.com/avehtari/rhat_ess/blob/master/code/monitornew.R

Lines: 106+
```
acov <- lapply(seq_len(chains), function(i) autocovariance(sims[, i])) 
acov <- do.call(cbind, acov)
rho_hat_odd <- 1 - (mean_var - mean(acov[t + 2, ])) / var_plus
```
but is done for the first part

`mean_var <- mean(acov[1, ]) * n_samples / (n_samples - 1) `

Reply:
@avehtari do we have a bug in here?

Reply:
@ahartikainen can you clarify? (you can also email or slack me, although I probably don't answer until tomorrow)

Reply:
in monitor.R

```
ess_rfun <- function(sims) {
...
  #### acov create ###
  acov <- lapply(seq_len(chains), function(i) autocovariance(sims[, i])) 
  acov <- do.call(cbind, acov)
  chain_mean <- apply(sims, 2, mean)
  #### first column in acov scaled with n_samples / (n_samples  - 1) (so basically 1/N-1)
  mean_var <- mean(acov[1, ]) * n_samples / (n_samples - 1) 
  #### rescaled back (so basically 1/N)
  var_plus <- mean_var * (n_samples - 1) / n_samples
  if (chains > 1) 
    var_plus <- var_plus + var(chain_mean)
  
  # Geyer's initial positive sequence
  rho_hat_t <- rep.int(0, n_samples)
  t <- 0
  rho_hat_even <- 1;
  rho_hat_t[t + 1] <- rho_hat_even
  #### here there is no scaling for the second column
  rho_hat_odd <- 1 - (mean_var - mean(acov[t + 2, ])) / var_plus
  rho_hat_t[t + 2] <- rho_hat_odd
  t <- 2  
  while (t < nrow(acov) - 1 && !is.nan(rho_hat_even + rho_hat_odd) && 
         (rho_hat_even + rho_hat_odd > 0)) {
    rho_hat_even = 1 - (mean_var - mean(acov[t + 1, ])) / var_plus
    rho_hat_odd = 1 - (mean_var - mean(acov[t + 2, ])) / var_plus
    if ((rho_hat_even + rho_hat_odd) >= 0) {
      rho_hat_t[t + 1] <- rho_hat_even
      rho_hat_t[t + 2] <- rho_hat_odd
    }
    t <- t + 2
  }
```

I know 1/N and 1/(N-1) don't really matter, just want to be certain should we scale the second column too?

btw, order could be simplified here (maybe?)

```
mean_var <- mean(acov[1, ]) * n_samples / (n_samples - 1) 
var_plus <- mean_var * (n_samples - 1) / n_samples
```
-->
```
var_plus <- mean(acov[1, ]) 
mean_var <- mean_var * n_samples / (n_samples - 1)
```